pkg/plugins/envvars/envvars.go

```diff
@@ -84,6 +85,26 @@ func (s *Service) tracingEnvVars(plugin *plugins.Plugin) []string {
 	return vars
 }
 
+func (s *Service) featureToggleEnableVar(ctx context.Context) []string {
+	var variables []string // an array is used for consistency and keep the logic simpler for no features case
+
+	if s.cfg.Features != nil {
+		enabledFeatures := s.cfg.Features.GetEnabled(ctx)
+
+		if len(enabledFeatures) > 0 {
+			features := make([]string, len(enabledFeatures))
+			i := 0
+			for feat := range enabledFeatures {
+				features[i] = feat
+				i++
+			}
```

Comment:
nit:

```suggestion
			for feat, i := range enabledFeatures {
				features[i] = feat
			}
```

Reply:
This suggestion won't work. `enabledFeatures` type is `map[string]bool`, therefore in the suggestion `i` would be `bool`.

Reply:
ah, right, right, ignore this. The other possibility is define slice with capacity and append instead:

```go
			features := make([]string, 0, len(enabledFeatures))
			for feat := range enabledFeatures {
					features = append(features, feat)
			}
```

but it's the same, just a matter of style, feel free to ignore.